src/ngraph_builder.cc

```diff
@@ -633,8 +633,152 @@ tf::Status Builder::TranslateGraph(const std::vector<tf::TensorShape>& inputs,
+      TF_RETURN_IF_ERROR(
+          tf::GetNodeAttr(op->attrs(), "dilations", &tf_dilations));
+      TF_RETURN_IF_ERROR(
+          tf::GetNodeAttr(op->attrs(), "padding", &tf_padding_type));
+      TF_RETURN_IF_ERROR(
+          tf::GetNodeAttr(op->attrs(), "data_format", &tf_data_format));
+
+      if (tf_data_format != "NHWC" && tf_data_format != "NCHW") {
+        return tf::errors::InvalidArgument(
+            "Conv2DBackpropInput data format is neither NHWC nor NCHW");
+      }
+
+      std::vector<tf::int64> tf_input_sizes;
+      TF_RETURN_IF_ERROR(tf::GetNodeAttr(op->attrs(),
+            "_ngraph_static_input_sizes", &tf_input_sizes));
+      if (std::any_of(tf_input_sizes.begin(), tf_input_sizes.end(),
+            [](tf::int32 size) { return size <= 0; })) {
+        return tf::errors::InvalidArgument(
+            "Conv2DBackpropInput input sizes must be positive integers");
+      }
+
+      bool is_nhwc = (tf_data_format == "NHWC");
+
+      NGRAPH_VLOG(3) << ng::join(tf_strides);
+      NGRAPH_VLOG(3) << ng::join(tf_dilations);
+      NGRAPH_VLOG(3) << tf_padding_type;
+      NGRAPH_VLOG(3) << tf_data_format;
+
+      ng::Strides ng_strides(2);
+      ng::Strides ng_dilations(2);
+      ng::Shape ng_image_shape(2);
+      ng::Shape ng_kernel_shape(2);
+      ng::Shape ng_batch_shape(4);
+
+      if (is_nhwc) {
+        ng_strides[0] = tf_strides[1];
+        ng_strides[1] = tf_strides[2];
+        ng_dilations[0] = tf_dilations[1];
+        ng_dilations[1] = tf_dilations[2];
+        ng_image_shape[0] = tf_input_sizes[1];
+        ng_image_shape[1] = tf_input_sizes[2];
+        ng_batch_shape = { static_cast<unsigned long>(tf_input_sizes[0]),
+                           static_cast<unsigned long>(tf_input_sizes[3]),
+                           static_cast<unsigned long>(tf_input_sizes[1]),
+                           static_cast<unsigned long>(tf_input_sizes[2]) };
+        auto& s = ng_out_backprop->get_shape();
+        ng::Shape reshaped{s[0], s[3], s[1], s[2]};
+        ng_out_backprop = make_shared<ng::op::Reshape>(
+            ng_out_backprop, ng::AxisVector{0, 3, 1, 2}, reshaped);
+      } else {
+        ng_strides[0] = tf_strides[2];
+        ng_strides[1] = tf_strides[3];
+        ng_dilations[0] = tf_dilations[2];
+        ng_dilations[1] = tf_dilations[3];
+        ng_image_shape[0] = tf_input_sizes[2];
+        ng_image_shape[1] = tf_input_sizes[3];
+        ng_batch_shape = { static_cast<unsigned long>(tf_input_sizes[0]),
+                           static_cast<unsigned long>(tf_input_sizes[1]),
+                           static_cast<unsigned long>(tf_input_sizes[2]),
+                           static_cast<unsigned long>(tf_input_sizes[3]) };
+      }
+
+      NGRAPH_VLOG(3) << "ng_strides: " << ng::join(ng_strides);
+      NGRAPH_VLOG(3) << "ng_dilations: " << ng::join(ng_dilations);
+      NGRAPH_VLOG(3) << "ng_image_shape: " << ng::join(ng_image_shape);
+
+      {
```

Comment:
Any specific reason for this new scope?

Reply:
it's copypasta from elsewhere, but I believe it's to not leak the `auto &s` outside, given that it's a bad but convenient variable name.

Reply:
I agree that there won't be any leak - but for reusing a variable name let's not introduce a new scope. This kind of scopes are best used for temporary locking/unlocking.

Reply:
The copypasta is from code I wrote. I can't remember why I did the new scope, to be honest.

Reply:
It's because you're reusing variable names, which makes sense to me in context.  Can we keep it?

Reply:
Oh! You're right. :) (Didn't spot/had forgotten the other use of `s` further up in the file.) I do agree that the nested scope makes sense here, but don't feel strongly about it.

Reply:
Yes - I would recommend removing the scope and renaming the variable :)

Reply:
It would then deviate from the rest of the code.  Can I fix this in refactoring instead?

Reply:
No - let's not combine the two.